barretenberg/cpp/src/barretenberg/honk/composer/ultra_composer.hpp

```diff
@@ -69,8 +72,24 @@ template <UltraFlavor Flavor> class UltraComposer_ {
     UltraProver_<Flavor> create_prover(std::shared_ptr<Instance>);
     UltraVerifier_<Flavor> create_verifier(std::shared_ptr<Instance>);
 
-    ProtoGalaxyProver_<Flavor> create_folding_prover(std::vector<std::shared_ptr<Instance>>);
-    ProtoGalaxyVerifier_<Flavor> create_folding_verifier(std::vector<std::shared_ptr<Instance>>);
+    ProtoGalaxyProver_<ProverInstances> create_folding_prover(std::vector<std::shared_ptr<Instance>> instances)
```

Comment:
After a lot of fighting linker and compile errors having the create folding prover and verifier functions in the header and specify the numbers of instances to be folded (see above) here seemed like the most reasonable thing to do. Very open to ways to improve this if there are any.